### PR TITLE
[topcrash] Specify the major version to avoid old builds

### LINKED ISSUE #1675: For topcrash, ignore very old versions
E.g. in https://bugzilla.mozilla.org/show_bug.cgi?id=1700242 there was a spike with a very old release version.
We can keep using the channel to ensure we are don't miss anything during the release week, but we should ignore very old versions.

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #1675

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
# Dry-run

<p>The following bugs are missing a topcrash keyword:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th><th>Severity</th><th>Added Keywords</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1802620">1802620</a>
</td>
<td>
Crash in [@ android.app.ForegroundServiceStartNotAllowedException: at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java)]
</td>
<td>
S2
</td>
<td>
topcrash
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1802046">1802046</a>
</td>
<td>
&#34;GeckoRuntime is shutting down&#34; Crash in [@ java.lang.RuntimeException: at org.mozilla.geckoview.WebExtensionController$$ExternalSyntheticLambda14.onShutdown(R8$$SyntheticClass:5)]
</td>
<td>
S3
</td>
<td>
topcrash
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1779558">1779558</a>
</td>
<td>
Crash in [@ mozalloc_abort | abort | _iris_batch_flush]
</td>
<td>
S2
</td>
<td>
topcrash
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1776917">1776917</a>
</td>
<td>
[Fission] Fenix tab crashes when attempting to access an unsecure page with HTTPS-Only Mode enabled
</td>
<td>
S3
</td>
<td>
topcrash
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1639157">1639157</a>
</td>
<td>
Crash while collecting the nursery
</td>
<td>
S4
</td>
<td>
topcrash-startup
</td>
</tr>
</tbody>
</table>
</p>

File: auto_nag/topcrash.py [hunk 1]
```diff
@@ -359,6 +361,7 @@ def __get_params_from_criterion(self, criterion: dict):
         params = {
             "product": criterion["product"],
             "release_channel": criterion["channel"],
+            "major_version": self._get_major_version_constrain(criterion["channel"]),
```
Comment (marco-c) [inline auto_nag/topcrash.py:364]: That's useful, I didn't know it existed!

File: auto_nag/topcrash.py [hunk 2]
```diff
@@ -379,6 +382,29 @@ def __get_params_from_criterion(self, criterion: dict):
...
         return params
 
+    def _get_major_version_constrain(self, channel: str) -> str:
+        """Return the major version constrain for the given channel."""
+        if self.__version_constrains is None:
+            versions = lmdversions.get(base=True)
+            last_release_date = lmdversions.getMajorDate(versions["release"])
+
+            if last_release_date > self.start_date:
+                # If the release date newer than the start date in the query, we
```
Comment (marco-c) [inline auto_nag/topcrash.py:392]:
```suggestion
                # If the release date is newer than the start date in the query, we
```

**Review by marco-c — DISMISSED** [review]

File: auto_nag/topcrash.py (hunk 1 above)
Comment (suhaibmujahid) [reply auto_nag/topcrash.py:364]: Me too, I found it when I was trying to experiment with Sebastian's suggestion :)

**Pull request merged**